core/cas-server-core/src/test/java/org/apereo/cas/DefaultCentralAuthenticationServiceTests.java

```diff
@@ -195,14 +195,26 @@ public void verifyGrantServiceTicketWithInvalidTicketGrantingTicket() throws Exc
     }
 
     @Test
-    public void verifyDelegateTicketGrantingTicketWithProperParams() {
-        val ctx = CoreAuthenticationTestUtils.getAuthenticationResult(getAuthenticationSystemSupport(), getService());
-        val ticketId = getCentralAuthenticationService().createTicketGrantingTicket(ctx);
-        val serviceTicketId = getCentralAuthenticationService().grantServiceTicket(ticketId.getId(), getService(), ctx);
-        val ctx2 = CoreAuthenticationTestUtils.getAuthenticationResult(getAuthenticationSystemSupport(),
-            RegisteredServiceTestUtils.getHttpBasedServiceCredentials());
-        val pgt = getCentralAuthenticationService().createProxyGrantingTicket(serviceTicketId.getId(), ctx2);
-        assertNotNull(pgt);
+    public void verifyDelegateTicketGrantingTicketWithProperParamsWithRemoteTicketRegistry() {
```

Comment:
I understand what is happening here, but I think it's a lot better if you actually provided a test with a real ticket registry like memcached. Simulating this is not 100% accurate, and may not be maintainable.

For example, you'd create a test case inside the memcached module that auto-creates the memcached ticket registry, and then invokes operations on this "CAS" class. This way, you don't have to simulate, and you don't have to "set" the ticket registry, and you rely on the auto-configuration to create registries for you and inject those. This is much closer to the runtime, and there is no need to simulate anything, since you're dealing with a real registry. 

If necessary, the same test can then be applied to other registries such as Hazelcast, etc.

Reply:
OK. I loved my `SimulateRemoteTicketRegistry`, but I was hesitating. I will create a real test in the Memcached module.

Reply:
I will of course admit that this is quite clever. Nicely done! 😄

Reply:
I just added a Memcached test to confirm the fix.